onnxruntime/core/graph/ep_api_types.cc

```diff
@@ -0,0 +1,541 @@
+  consumer_infos.reserve(nodes.size());
+  for (const Node* node : nodes) {
+    const EpNode* ep_node = graph_->index_to_ep_node.GetEpNode(node->Index());
+    if (ep_node == nullptr) {
+      continue;  // Node is not in this GraphViewer
+    }
+
+    std::vector<int64_t> input_indices;
+    ORT_RETURN_IF_ERROR(GetInputIndices(*ep_node, name_, input_indices));
+
+    for (int64_t input_index : input_indices) {
+      OrtValueInfo::ConsumerInfo use_info(ep_node->ToExternal(), input_index);
+      consumer_infos.push_back(use_info);
+    }
+  }
+
+  result = std::move(consumer_infos);
+  return Status::OK();
+#else
+  ORT_UNUSED_PARAMETER(result);
+  return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED,
+                         "Getting the consumers of an OrtValueInfo is not supported in this build");
+#endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+}
+
+Status EpValueInfo::GetNumConsumerInfos(size_t& num_consumers) const {
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  num_consumers = 0;
+
+  if (graph_ == nullptr) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Unable to get number of uses of OrtValueInfo '", name_,
+                           "' that is not owned by a OrtGraph.");
+  }
+
+  std::vector<const Node*> nodes = graph_->graph_viewer.GetConsumerNodes(name_);
+  if (nodes.empty()) {
+    return Status::OK();
+  }
+
+  for (const Node* node : nodes) {
+    const EpNode* ep_node = graph_->index_to_ep_node.GetEpNode(node->Index());
+    if (ep_node == nullptr) {
+      continue;  // Node is not in this GraphViewer
+    }
+
+    std::vector<int64_t> input_indices;
+    ORT_RETURN_IF_ERROR(GetInputIndices(*ep_node, name_, input_indices));
+
+    num_consumers += input_indices.size();  // A single OrtNode can use an OrtValueInfo as an input more than once.
+  }
+
+  return Status::OK();
+#else
+  ORT_UNUSED_PARAMETER(num_consumers);
+  return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED,
+                         "Getting the consumers of an OrtValueInfo is not supported in this build");
+#endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+}
+
+Status EpValueInfo::GetInitializerValue(const OrtValue*& result) const {
+  if (!IsFlagSet(kIsInitializer)) {
+    // This OrtValueInfo does not represent an initializer. Set result to nullptr and return an OK status
+    // to allow user to use this function to check if this is an initializer.
+    result = nullptr;
+    return Status::OK();
+  }
+
+  const EpGraph* curr_graph = graph_;
+
+  while (curr_graph != nullptr) {
+    auto map_iter = curr_graph->initializer_values.find(name_);
+
+    if (map_iter != curr_graph->initializer_values.end()) {
+      result = map_iter->second.get();
+      return Status::OK();
+    }
+
+    const EpNode* parent_node = curr_graph->parent_node;
+    curr_graph = parent_node != nullptr ? parent_node->ep_graph : nullptr;
```

Comment:
nit: just wondering whether there is a case that needs to return nullptr if the EpValueInfo is outer scope initializer. 
If not, then the way outer scope EpValueInfo being handled in EpGraph::Create and being kept in this level of EpGraph is nice.

Reply:
Actually, there's a way to allow this to return a valid value even when the graphs are traversed bottom-up. I wasn't properly accounting for GetCapabilities traversal order for subgraphs.

Reply:
Updated this to always return a valid OrtValue for "outer scope" initializers even if the graph is traversed bottom-up. Please let me know if I missed anything.